### PR TITLE
Fix filter bug, when hipcc/nvcc/clang-cuda was defined and the hip/cuda backend was not defined

### PR DESCRIPTION
**Pull request by SimeonEhrig**
For hipcc and nvcc/clang-cuda there was the bug, that if the hip/cuda back-end was not explicit defined and other back-ends was defined, it passes the back-end filter.

### CODE REVIEW

File: tests/test_cuda_sdk.py [hunk 1]
```diff
@@ -294,6 +294,21 @@ def test_backend_versions(self):
             ]
             self.assertFalse(compiler_backend_filter_typed(comb4))
 
+        # the filter should return false, when other backends than the
+        # ALPAKA_ACC_GPU_CUDA_ENABLE are defined
```
Comment (j-stephan) [inline tests/test_cuda_sdk.py:298]:
```suggestion
        # the filter should return false when back-ends other than
        # ALPAKA_ACC_GPU_CUDA_ENABLE are defined
```

File: tests/test_cuda_sdk.py [hunk 2]
```diff
@@ -765,6 +780,17 @@ def test_host_device_compiler_backend(self):
                     (ALPAKA_ACC_GPU_HIP_ENABLE, "5.0"),
                 ],
             ],
+            # the filter should return false, when other backends than the
+            # ALPAKA_ACC_GPU_CUDA_ENABLE are defined
```
Comment (j-stephan) [inline tests/test_cuda_sdk.py:784]:
```suggestion
            # the filter should return false when back-ends other than
            # ALPAKA_ACC_GPU_CUDA_ENABLE are defined
```

File: tests/test_hipcc.py [hunk 3]
```diff
@@ -271,6 +271,16 @@ def test_host_device_compiler_backend(self):
                     (ALPAKA_ACC_GPU_HIP_ENABLE, OFF_VER),
                 ],
             ],
+            # the filter should return false, when other backends than the
+            # ALPAKA_ACC_GPU_HIP_ENABLE are defined
```
Comment (j-stephan) [inline tests/test_hipcc.py:275]:
```suggestion
            # the filter should return false when back-ends other than
            # ALPAKA_ACC_GPU_HIP_ENABLE are defined
```

File: tests/test_hipcc.py [hunk 4]
```diff
@@ -314,3 +324,13 @@ def test_host_device_compiler_backend(self):
                 f"DEVICE_COMPILER: {comb[1][0]} {comb[1][1]}, "
                 f"BACKENDS: {comb[2]}",
             )
+
+
+global param_map
+# set param_map, that filters expect the following parameters in the
+# order
```
Comment (j-stephan) [inline tests/test_hipcc.py:331]: Could you rephrase this? I don't really understand what you are trying to say here.

**Review by j-stephan — CHANGES_REQUESTED** [review]
Just a few minor comments:

File: tests/test_hipcc.py (hunk 4 above)
Comment (SimeonEhrig) [reply tests/test_hipcc.py]: This was a left-over from debugging. I removed the code block.

**Comment by SimeonEhrig** [comment]
@j-stephan Ready for merge
